server.js

```diff
@@ -37,14 +41,26 @@ app.prepare().then(() => {
   // This is useful for fully local development.
   server.use('/test', proxy({ target: BACKEND_URL, changeOrigin: true }));
 
-  server.get('/', (req, res) => res.redirect(`/${DEFAULT_LOCALE}/`));
+  server.get('/', (req, res) => res.redirect(`/${getUserLang(req)}/`));
+
   server.get('/:startPath*', (req, res) => {
     const { startPath } = req.params;
-    if (VALID_PATHS.includes(startPath)) {
+    const userLang = getUserLang(req);
+
+    if (STATIC_PATHS.includes(startPath)) {
       return handle(req, res);
     }
-    // missed locale, add `be` & redirect
-    return res.redirect(`/${DEFAULT_LOCALE}${req.originalUrl}`);
+
+    if (VALID_LOCALES.includes(startPath)) {
+      if (userLang === startPath || !hasLangCookie(req)) {
+        return handle(req, res);
+      }
+
+      return res.redirect(`/${userLang}${req.params['0']}`);
```

Comment:
what is `req.params['0']`?

Reply:
It is a part of the url that follows startPath. In the case above, req.params['0'] would be equal to '/about'.